subprojects/gradle/ci-logger/src/main/kotlin/com/avito/utils/logging/ElasticDestination.kt

```diff
@@ -1,15 +1,30 @@
 package com.avito.utils.logging
 
 import com.avito.android.elastic.ElasticClient
+import com.avito.android.elastic.ElasticConfig
+import com.avito.android.elastic.ElasticFactory
 
 internal class ElasticDestination(
-    private val elasticClient: ElasticClient,
+    private val config: ElasticConfig,
     private val tag: String,
     private val level: String
 ) : CILoggingDestination {
 
+    @Transient
+    private lateinit var _client: ElasticClient
```

Comment:
I prefer to have one cached instance on the static level.
The current decision will produce a lot of instances without OkHttpClient sharing. As I know OkHttpClient is heavy.

Reply:
We could use `gradle.serviceOf<>`

Reply:
Yes, this might be a good optimization for the whole CILogger. I'll make an issue for it.